inst/examples-shiny/toast/app.R

```diff
@@ -0,0 +1,481 @@
+        actionButton("hide_toast", "Hide Last Toast", class = "btn-secondary")
+      )
+    ),
+
+    layout_column_wrap(
+      width = 1,
+
+      # Advanced Features and Examples
+      card(
+        card_header("Advanced Features"),
+        card_body(
+          actionButton(
+            "show_persistent",
+            "Show Persistent Toast",
+            class = "mb-2 w-100"
+          ),
+          actionButton(
+            "hide_persistent",
+            "Hide Persistent Toast",
+            class = "mb-2 w-100"
+          ),
+          actionButton(
+            "show_long_duration",
+            "Long Duration (10s)",
+            class = "mb-2 w-100"
+          ),
+          actionButton(
+            "show_no_close",
+            "No Close Button",
+            class = "mb-2 w-100"
+          ),
+          actionButton(
+            "show_custom_header",
+            "Custom Header with Icon & Status",
+            class = "mb-2 w-100"
+          )
+        )
+      ),
+
+      card(
+        card_header("Interactive Toasts"),
+        card_body(
+          actionButton(
+            "show_action_buttons",
+            "Toast with Action Buttons",
+            class = "mb-2 w-100"
+          ),
+          actionButton(
+            "show_multiple",
+            "Show Multiple Toasts",
+            class = "mb-2 w-100"
+          ),
+          actionButton(
+            "show_all_positions",
+            "Test All Positions",
+            class = "mb-2 w-100"
+          ),
+          actionButton(
+            "show_dynamic_content",
+            "Toast with Dynamic Content",
+            class = "mb-2 w-100"
+          )
+        )
+      )
+    )
+  )
+)
+
+server <- function(input, output, session) {
+  # Store last toast ID
+  last_toast_id <- reactiveVal(NULL)
+  persistent_toast_id <- reactiveVal(NULL)
+
+  # Show toast from builder
+  observeEvent(input$show_toast, {
+    # Build header if needed
+    header <- NULL
+    if (input$use_header) {
+      header <- toast_header(
+        title = input$header_title,
+        icon = if (nzchar(input$header_icon)) {
+          bsicons::bs_icon(input$header_icon)
+        },
+        status = if (nzchar(input$header_status)) input$header_status
+      )
+    }
+
+    body_icon <- if (!input$use_header && nzchar(input$icon_body)) {
+      bsicons::bs_icon(input$icon_body)
+    }
+
+    # Build toast
+    toast_obj <- toast(
+      input$body,
+      header = header,
+      icon = body_icon,
+      id = if (nzchar(input$custom_id)) input$custom_id,
+      type = if (nzchar(input$type)) input$type,
+      duration_s = input$duration_s,
+      position = input$position,
+      closable = input$closable
+    )
+
+    # Show and store ID
+    id <- show_toast(toast_obj)
+    last_toast_id(id)
+  })
+
+  # Hide last toast
+  observeEvent(input$hide_toast, {
+    req(last_toast_id())
+    hide_toast(last_toast_id())
+    last_toast_id(NULL)
+  })
```

Comment:
If I click 'show' repeatedly, then 'hide' repeatedly, only the 1st hide works (since we're only tracking the last ID).

Related to this problem, would it make sense for `hide_toast()` to remove the "oldest" toast? Or provide access to a `input[["toast_id"]] == T/F` ?